### PR TITLE
INTMDB-567: Add shellcheck as Github action

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

This PR adds shellchek as GitHub action.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments
Next Steps:

- There are currently a lot of copy/paste scripts all around the resources. We will investigate which scripts are really needed and delete the others.

### CODE REVIEW

File: cfn-resources/cfn-submit-helper.sh [hunk 1]
```diff
@@ -15,15 +15,17 @@
...
+# shellcheck source=/dev/null
 . ./cfn-submit-helper.config
 env | grep CFN_SUBMIT_
 #echo "AWS_DEFAULT_PROFILE=${AWS_DEFAULT_PROFILE}"
 
 _DRY_RUN=${DRY_RUN:-false}
 _BUILD_ONLY=${BUILD_ONLY:-false}
 _SUBMIT_ONLY=${SUBMIT_ONLY:-false}
-_CLOUD_PUBLISH=${CLOUD_PUBLISH:-false}
+#_CLOUD_PUBLISH=${CLOUD_PUBLISH:-false}
```
Comment (gssbzn) [inline cfn-resources/cfn-submit-helper.sh:28]: why commenting this?

Comment (andreaangiolillo) [reply cfn-resources/cfn-submit-helper.sh:28]: shellscript was complaining about `_CLOUD_PUBLISH` not being used. I am not 100% sure why is there and I want to wait to discuss all these scripts with consultants before refactoring them

File: .github/cleanup-runs-hub.sh [hunk 2]
```diff
@@ -4,7 +4,7 @@ echo "cleanup-runs-hub.sh"
 echo "About: Deletes all the runs from a Github action."
 
 REPO=$1
-hub api /repos/$REPO/actions/runs | \
+hub api /repos/"$REPO"/actions/runs | \
```
Comment (gssbzn) [inline .github/cleanup-runs-hub.sh:7]: why not

```suggestion
hub api "/repos/$REPO/actions/runs" | \
```

File: cfn-resources/cfn-publish.sh [hunk 3]
```diff
@@ -1,31 +1,26 @@
...
 do
 paramKey="${param}="
-paramValue=$(echo $otherParams | jq -c -r --arg key $param '.[$key]')
-exportString=$(echo "$paramKey$paramValue")
-export "${exportString}"
+paramValue=$(echo "$otherParams" | jq -c -r --arg key "$param" '.[$key]')
+exportString="$paramKey$paramValue"
+export exportString
 echo
 done
```
Comment (gssbzn) [inline cfn-resources/cfn-publish.sh:22]: since you are here can you fix this identation

File: cfn-resources/cfn-submit-helper.sh (hunk 1 above)
Comment (gssbzn) [reply cfn-resources/cfn-submit-helper.sh:28]: it probably makes sens then to delete if unused

File: cfn-resources/cloud-backup-snapshot-export-bucket/test/cfn-test-create-inputs.sh [hunk 4]
```diff
@@ -54,36 +55,37 @@ echo cat add-policy.json
...
+echo "--------------------------------authorize mongodb  Role ends ----------------------------"
 
 #create the s3 bucket
 
 bucketName="cloud-backup-snapshot-test123-"${awsRegion}
 
-aws s3 rb s3://${bucketName} --force
-aws s3 mb s3://${bucketName} --output json
+aws s3 rb s3://"${bucketName}" --force
+aws s3 mb s3://"${bucketName}" --output json
```
Comment (gssbzn) [inline cfn-resources/cloud-backup-snapshot-export-bucket/test/cfn-test-create-inputs.sh:88]:
```suggestion
aws s3 rb "s3://${bucketName}" --force
aws s3 mb "s3://${bucketName}" --output json
```

**Review by gssbzn — APPROVED** [review]
LGTM comments are not blocking

File: cfn-resources/cfn-publish.sh (hunk 3 above)
Comment (andreaangiolillo) [reply cfn-resources/cfn-publish.sh]: Updated. I added [mvdan.cc/sh/v3/cmd/shfmt@latest ](https://github.com/mvdan/sh)to the devtolls and indented all the sh files. I will add this tool to the cli as well xD

File: .github/cleanup-runs-hub.sh (hunk 2 above)
Comment (andreaangiolillo) [reply .github/cleanup-runs-hub.sh]: Updated

File: cfn-resources/cloud-backup-snapshot-export-bucket/test/cfn-test-create-inputs.sh (hunk 4 above)
Comment (andreaangiolillo) [reply cfn-resources/cloud-backup-snapshot-export-bucket/test/cfn-test-create-inputs.sh]: Updated

File: cfn-resources/cfn-submit-helper.sh (hunk 1 above)
Comment (andreaangiolillo) [reply cfn-resources/cfn-submit-helper.sh]: updated

**Pull request merged**